src/solvers/flattening/boolbv_byte_update.cpp

```diff
@@ -25,14 +25,7 @@ bvt boolbvt::convert_byte_update(const byte_update_exprt &expr)
   const exprt &offset_expr=expr.offset();
   const exprt &value=expr.value();
 
-  bool little_endian;
-
-  if(expr.id()==ID_byte_update_little_endian)
-    little_endian=true;
-  else if(expr.id()==ID_byte_update_big_endian)
-    little_endian=false;
-  else
-    UNREACHABLE;
+  const bool little_endian = expr.id() == ID_byte_update_little_endian;
```

Comment:
I assume the invariant that `id()` can only have two values here has already been checked higher up.

Reply:
I have added a `PRECONDITION` to that effect.